### PR TITLE
Specify holidays for each invocation

### PR DESCRIPTION
**Pull request by cadenforrest**
This PR implements the feature and fixes the merge conflicts in https://github.com/bokmann/business_time/pull/180. The original author has been unresponsive since opening the PR, so I can take it from here if further changes are requested. 

I'd like to credit the original author however is appropriate, so let me know if there's something I should do outside of listing them in the contributor list. 

Closes #180

### CODE REVIEW

File: lib/business_time/time_extensions.rb [hunk 1]
```diff
@@ -194,5 +197,34 @@ def consecutive_days
...
+              when Date
+                array_of_dates << holiday
+              else
+                raise ArgumentError.new("Holidays specified must be a Date, Time or String.")
+            end
+          end
+      end
+
+      array_of_dates
+    end
```
Comment (rmm5t) [inline lib/business_time/time_extensions.rb:203]: [SUGGESTION] I think this entire `#to_array_of_dates` method can be distilled down to something much simpler. For example:

```ruby
def to_array_of_dates(values)
  Array.wrap(values).map(&:to_date)
end
```

`Array.wrap` wraps its arg in an array unless it's already an array.

`String`, `Time`, and `Date` all respond to `#to_date` (thanks to ActiveSupport).   If someone passes something that doesn't quack with the `#to_date` method, let the `NoMethodError` bubble up and ignore the `ArgumentError` check.

**Review by rmm5t — CHANGES_REQUESTED** [review]
Thank you for this effort. Given that @bokmann [was interested in this during the first PR](https://github.com/bokmann/business_time/pull/180#issuecomment-390290354), I think we can get this merged in.  This might be the last version bump before a v1.0 release, or this might be the feature that takes us to a v1 release.

I have one code suggestion. Please consider that code change, and get back to me. 

With regards to credit, if you could add an entry to the "Unreleased" section of the `CHANGELOG` pointing at both PRs, that would be helpful too.

File: lib/business_time/time_extensions.rb (hunk 1 above)
Comment (cadenforrest) [reply lib/business_time/time_extensions.rb:203]: @rmm5t Very cool, just committed the changes 👍

**Review by rmm5t — APPROVED** [review]

**Review by rmm5t — APPROVED** [review]

**Comment by rmm5t** [comment]
Released in v0.13.0
https://rubygems.org/gems/business_time

Thank you!

**Pull request merged**